### ISSUE TITLE
Validate incorrectly enforces file naming requirements on bundles/collections

### ISSUE THREAD

**Issue by jordanpadams** [NASA-PDS-Incubator/validate#561]
## 🐛 Describe the bug

Earlier versions of the PDS4 Standards References included file naming guidelines for bundles/collections that were interpreted as requirements.

Validate needs to be updated to execute correctly without using those file naming assumptions.

SR 6C.1.3 mentions the following as reserved file names:
```
bundle*.[xml,lblx]
collection*.[xml,lblx,csv]
```

But nowhere else in the SR does it actually require bundle/collection be named with that pattern. Anywhere this is assumed needs to be ripped out of validate.

## 📜 To Reproduce
1. Run validate with collection rule on the test data below:
```
validate --rule pds4.collection --target ~/Downloads/weirdError/calibration_files/ --label-extension lblx --skip-context-validation
```
2. See that integrity failures are thrown for all products in the collection.
4. Rename `calibration_files.lblx` to `calibration_files_collection.lblx`
5. See that it now validates successfully (with a few warnings).
6. Should be able to see this with any bundle/collection when trying to run validate with differing file names.

## 🕵️ Expected behavior

Validate will crawl and reads all files the same, and then determines the handling of those products by their product type.

## 📚 Version of Software Used

v3.0.3

## 🩺 Test Data / Additional context

[weirdError.zip](https://github.com/NASA-PDS/validate/files/10108365/weirdError.zip)

---

**Comment by smclaughlin7** [comment]
Hi @jordanpadams!  Just want to make sure we/NSSDCA understand this ticket:

1. When this bug is fixed will Validate no longer require bundle and collection file names to contain the strings "bundle" and "collection", respectively? 
2. Does Validate open bundle and collection labels to identify the product type?

Thanks!

**Comment by jordanpadams** [comment]
@smclaughlin7 

> 1. When this bug is fixed will Validate no longer require bundle and collection file names to contain the strings "bundle" and "collection", respectively?

correct.

> 2. Does Validate open bundle and collection labels to identify the product type?

right now, it assumes if it says "bundle" or "collection" it is one or the other. but eventually we do open the labels and the software would break down if they "lied" about that. after implementing this, we will no longer care, it will open every labels and process based upon the product type

**Comment by smclaughlin7** [comment]
@jordanpadams  Thanks!  Regarding

> > 1. When this bug is fixed will Validate no longer require bundle and collection file names to contain the strings "bundle" and "collection", respectively?
> 
> correct.

I need to investigate if this will affect our NSSDCA processes, which may look for the strings "bundle" and "collection" in the SIP manifest because accouding to 
PDS Standards Reference **2B.2.2.2 Subdirectories** https://pds.nasa.gov/datastandards/documents/sr/current/StdRef_1.18.0.:
Beneath the root directory in this example structure, the top-level subdirectories have a one-to-
one correspondence with the bundle’s collections. **_Each collection subdirectory has a collection
label with a name of the form “collection[_*] and having an extension of either “.xml” or “.lblx”.
and a collection inventory table with a name of the form “collection[_*].csv”. (Sometimes
“collection[_*]_inventory.csv” is used.)_**  See Section 9C.1 for details on construction of
collection inventory tables.

Might this be a topic for the DDWG?

**Comment by smclaughlin7** [comment]
@jordanpadams Oops!  Forgot to add this paragraph from **SR 2B.2.2.2 Subdirectories** about the bundle file name:
**_A bundle is described by an XML label with the name “bundle[_*] and having an extension of
either “.xml” or “.lblx”._** One or more of these files may be included under the root of the transfer
directory structure, but none is required

**Comment by jordanpadams** [comment]
@smclaughlin7 I guess so. I didn't see this portion of the standards reference. If nothing else, the documentation needs to be updated to include that text in a more appropriate location (e.g. where the collections and bundles are defined). It almost seems like this was pulled out before, and missed down in that part of the SR.

@mdrum can we add this to the list ASAP? This was an issue Anne raised.

**Comment by smclaughlin7** [comment]
@jordanpadams @mdrum  Thank you for considering this topic for the DDWG.

I was not aware of removing bundle/collection file naming constraints from the SR.  One thought:  If the "bundle" and "collection" strings are no longer required in the filenames, then end user --  a researcher, NSSDCA, etc. -- may need to open every label in a data collection to determine which items are data products and which one is the collection product.

**Comment by mdrum** [comment]
Are we sure this requirement was ever removed? I don't recall a ticket indicating such

**Comment by jordanpadams** [comment]
@mdrum it wasn't removed explicitly, but this requirement just seems appear vague (and in the wrong place) in the SR. if we want it to still be in there, then the SR should be updated to more clearly state this when bundles / collections are defined.

@smclaughlin7 on your point, agreed on the file system browse perspective. however, in the end, bundles/collections are really just groupings of products together. Our clients/interfaces should really be providing a more clear way to get to the products directly, and shouldn't really care about the bundle/collection files themselves.

**Comment by smclaughlin7** [comment]
@jordanpadams @mdrum Good news. Our NSSDCA software dev team confirmed our PDS4 processes should not be affected if this issue is implememnted.  (The processes do not look for "bundle" and "collection" filename strings when working on a Deep Archive SIP manifest.)

**Comment by jordanpadams** [comment]
since this is only a issue for a small number of data providers, tabling this for time being. other bigger bugs/features need attention

**Comment by jordanpadams** [comment]
status: in work. significant code changes expected

**Comment by al-niessner** [comment]
Forgot closes statement on #637 that fixed this issue.
